### ISSUE TITLE
`nomad run` should return non-zero exit code in case of any errors

### ISSUE THREAD

**Issue by pznamensky** [hashicorp/nomad#11082]
### Proposal
Return non-zero exit code in case of any errors during deploying an app.

### Use-cases

We switched to `nomad run` from `levant deploy` in our CI scripts, but we noticed, that `nomad run` can return code 0 even when  a deployment is failed:
```
~ $ nomad run my-app.nomad
...
2021-08-25T16:53:10+03:00
ID          = 098fb4b6
Job ID      = my-app
Job Version = 0
Status      = failed
Description = Failed due to progress deadline

Deployed
Task Group  Desired  Placed  Healthy  Unhealthy  Progress Deadline
my-app         1        1       0        1          2021-08-25T16:53:10+03:00

~ $ echo $?
0
```

### Attempted Solutions
Return non-zero code in case of any failures :)

**Comment by goku321** [comment]
I'm new to nomad but have experience with Go and k8s. Can I work on this?

**Comment by mikenomitch** [comment]
Hey @goku321, please feel free to take this one. We would love to see a community PR for this!

Because this changes the external interface for the command, I think we might wait to implement this until a major release, but I think this is a good idea.

Good luck!

**Comment by goku321** [comment]
Thank you, @mikenomitch

**Comment by tgross** [comment]
Fixed in #11550 and will ship in the next planned patch release.
